client/asset/dcr/rpcwallet.go

```diff
@@ -580,6 +656,8 @@ func (w *rpcWallet) SyncStatus(ctx context.Context) (bool, float32, error) {
 // AddressPrivKey fetches the privkey for the specified address.
 // Part of the Wallet interface.
 func (w *rpcWallet) AddressPrivKey(ctx context.Context, address stdaddr.Address) (*dcrutil.WIF, error) {
+	w.rpcMtx.RLock()
+	defer w.rpcMtx.RUnlock()
```

Comment:
Maybe this mutex is just overly cautious.  It's not clear if there would ever be any of these normal method calls concurrent with a Connect or Disconnect call, so maybe this is solvable with documentation.